parameter/MappingContext.cpp

```diff
@@ -28,113 +28,59 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 #include "MappingContext.h"
-#include <assert.h>
-#include <string.h>
+#include <cassert>
+#include <algorithm>
 #include <stdlib.h>
 
-using std::string;
-
-CMappingContext::CMappingContext(size_t uiNbItemTypes) : _pstItemArray(new CMappingContext::SItem[uiNbItemTypes]), _uiNbItemTypes(uiNbItemTypes)
-{
-    // Clear items
-    memset(_pstItemArray, 0, sizeof(*_pstItemArray) * uiNbItemTypes);
-}
-
-CMappingContext::~CMappingContext()
-{
-    delete [] _pstItemArray;
-}
-
-// Copy constructor
-CMappingContext::CMappingContext(const CMappingContext& from) : _pstItemArray(new CMappingContext::SItem[from._uiNbItemTypes]), _uiNbItemTypes(from._uiNbItemTypes)
-{
-    // FIXME: delegate constructor
-    // Copy content items
-    memcpy(_pstItemArray, from._pstItemArray, sizeof(*_pstItemArray) * _uiNbItemTypes);
-}
-
-// Affectation
-CMappingContext& CMappingContext::operator=(const CMappingContext& right)
-{
-    // FIXME: use a vector of CMappingContext::SItem
-    if (&right != this) {
-
-        // Size
-        _uiNbItemTypes = right._uiNbItemTypes;
-
-        // Content
-        // Delete previous array
-        delete [] _pstItemArray;
-
-        // Reallocate it
-        _pstItemArray = new CMappingContext::SItem[_uiNbItemTypes];
-
-        // Copy content items
-        memcpy(_pstItemArray, right._pstItemArray, sizeof(*_pstItemArray) * _uiNbItemTypes);
-    }
-    return *this;
-}
+using namespace std;
```

Comment:
We try and avoid to do this. You only seem to need `string`, `find_if` (only once - no need for a `using`), `end` and `begin`.